### PR TITLE
Re-introduced code analysers and fixed many violations

### PR DESCRIPTION
**Pull request by JR-Morgan**
# Notable breaking changes
- `[Obsolete(true)]` several old GraphQL functions, most notably subscriptions since they were very smelly
- Removed manager API stuff... its very smelly, is there a chance that we'll regret this?
- `BaseObjectDeserialzierV2` is now disposable
- Deleted `Helpers.State`, `Models.ApplicationObject`, `Helpers.Receive`

To check with @adamhathcock 
Are the changes to `GetHashCode` ok?...

### CODE REVIEW

File: src/Speckle.Sdk/Common/HashCode.cs [hunk 1]
```diff
@@ -7,21 +7,24 @@ namespace Speckle.Sdk.Common;
...
-  private readonly int value;
+  private const int EMPTY_COLLECTION_PRIME_NUMBER = 19;
+  private readonly int _value;
 
   /// <summary>
   /// Initializes a new instance of the <see cref="HashCode"/> struct.
   /// </summary>
   /// <param name="value">The value.</param>
-  private HashCode(int value) => this.value = value;
+  private HashCode(int value)
```
Comment (adamhathcock) [inline src/Speckle.Sdk/Common/HashCode.cs:17]: why drop the short constructors?  just a style thing

File: src/Speckle.Sdk/Common/HashCode.cs [hunk 2]
```diff
@@ -130,9 +127,14 @@ private static int GetHashCode<T>(IEnumerable<T> items, int startHashCode)
...
     else
     {
-      temp = CombineHashCodes(temp, EmptyCollectionPrimeNumber);
+      temp = CombineHashCodes(temp, EMPTY_COLLECTION_PRIME_NUMBER);
     }
 
     return temp;
   }
+
+  public int ToInt32()
```
Comment (adamhathcock) [inline src/Speckle.Sdk/Common/HashCode.cs:136]: what's this for?  the comment says don't do it 😬

File: src/Speckle.Sdk/Common/HashCode.cs (hunk 1 above)
Comment (JR-Morgan) [reply src/Speckle.Sdk/Common/HashCode.cs:17]: I may not have intended to change this...
I don't really mind which syntax is used

File: src/Speckle.Sdk/Common/HashCode.cs (hunk 2 above)
Comment (JR-Morgan) [reply src/Speckle.Sdk/Common/HashCode.cs:136]: There is a code analyser that is asking for this: https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca2225

Operator overloads are not the most convenient to call from other .NET languages, or pass around as delegates etc..

Comment (JR-Morgan) [reply src/Speckle.Sdk/Common/HashCode.cs:136]: And another analyser that says that `GetHashCode` functions should never throw https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca1065

Hence my other related changes.
I can't see any problems with implementing the way I have done.

File: src/Speckle.Sdk/Common/HashCode.cs (hunk 1 above)
Comment (JR-Morgan) [reply src/Speckle.Sdk/Common/HashCode.cs]: reverted constructor, and made [ide0021](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/style-rules/ide0021) a suggestion

**Review by adamhathcock — APPROVED** [review]
LGTM without going in to real detail.  Style is good

**Pull request merged**